File: scripts/graphcraft/assets/packages/ui-core/flutter/lib/screens/login_screen.dart
import 'package:flutter/material.dart';

import '../components/button_primary.dart';
import '../tokens.dart';

/// Login screen — design/screens/login.example.yaml
/// @graphcraft implements screen:login
class LoginScreen extends StatelessWidget {
  const LoginScreen({super.key, this.onLogin, this.onForgotPassword});

  final VoidCallback? onLogin;
  final VoidCallback? onForgotPassword;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: DesignTokens.colorBgDefault,
      body: SafeArea(
        child: Padding(
          padding: EdgeInsets.all(DesignTokens.spacingScreenPadding),
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.start,
            children: [
              Text(
                'Login',
                style: TextStyle(
                  color: DesignTokens.colorTextPrimary,
                  fontSize: 24,
                  fontWeight: FontWeight.w600,
                ),
              ),
              SizedBox(height: DesignTokens.spacingScreenPadding),
              ButtonPrimary(label: 'Sign in', onPressed: onLogin),
              const SizedBox(height: 12),
              ButtonPrimary(label: 'Forgot password', onPressed: onForgotPassword),
            ],
          ),
        ),
      ),
    );
  }
}

File: packages/ui-core/unity/Runtime/Screens/LoginScreen.cs
using UnityEngine;
using UnityEngine.UI;

namespace GraphCraft.UI
{
    /// <summary>
    /// Login meta-UI screen — design/screens/login.example.yaml
    /// @graphcraft implements screen:login
    /// </summary>
    public class LoginScreen : MonoBehaviour
    {
        public ButtonPrimary signInButton;
        public ButtonPrimary forgotPasswordButton;

        void Start()
        {
            ApplySafeAreaPadding();

            if (signInButton != null)
            {
                signInButton.SetLabel("Sign in");
            }
            if (forgotPasswordButton != null)
            {
                forgotPasswordButton.SetLabel("Forgot password");
            }
        }

        void ApplySafeAreaPadding()
        {
            var padding = Screen.safeArea;
            var rect = GetComponent<RectTransform>();
            if (rect == null)
            {
                return;
            }
            rect.offsetMin = new Vector2(padding.xMin, padding.yMin);
            rect.offsetMax = new Vector2(-(Screen.width - padding.xMax), -(Screen.height - padding.yMax));
        }
    }
}
